src/tools/knowledge_graph_tools.py

```
@@ -0,0 +1,586 @@
+
+        # Validate script path
+        validation = _validate_script_path(script_path)
+        if not validation["valid"]:
+            raise ValidationError(validation["error"])
+
+        # Step 1: Analyze script structure using AST
+        analyzer = AIScriptAnalyzer()
+        analysis_result = analyzer.analyze_script(script_path)
+
+        if analysis_result.errors:
+            logger.warning(f"Analysis warnings for {script_path}: {analysis_result.errors}")
```

Pylint (3.3.7)

[warning] Use lazy % formatting in logging functions

(W1203)